plugin/kubernetes/kubernetes.go

```diff
@@ -110,7 +110,7 @@ func (k *Kubernetes) Services(ctx context.Context, state request.Request, exact
 
 		// Hard code the only valid TXT - "dns-version.<zone>"
```

Comment:
I think this check should happen before the `Records` check, so the more expensive Records check is only performed for TXT queries that do not match the static dns-version record.